### PR TITLE
Fix compilation with ghc 8.2.2, Support foreign-libs and sub-libs

### PR DESCRIPTION
**Pull request by OlivierSohn**
I couldn't test it though, see #35

### CODE REVIEW

**Comment by basvandijk** [comment]
Thanks for working on this!

I got the following when building so do bump the version bounds:

```
Setup: Encountered missing dependencies:
Cabal ==1.24.*, base >=4.5 && <4.10
```

After fixing that the package builds and runs successfully.

**Comment by OlivierSohn** [comment]
Ha, I was building with stack, and allow-newer = true :) I think it should be ok now

**Comment by basvandijk** [comment]
It works, thanks. Hopefully @hvr can merge and publish this soon.

**Comment by OlivierSohn** [comment]
The CI fails for GHC 8.0.2, it may be because it is using a too-recent version of Cabal :

```shell
compiled using version 2.3.0.0 of the Cabal library 
```

**Comment by hvr** [comment]
Build-failures like

```
packunused.hs:15:48: error:
    Module ‘Distribution.Package’ does not export ‘unUnitId'
```

means that the selected lib:Cabal version doesn't provide the expected API. I see `build-depends` still allows `Cabal-1.24.*`; should this rather be `build-depends: Cabal == 2.0.*` ?

**Comment by OlivierSohn** [comment]
@hvr yes, I should have changed the lower bound, I'll do that!

**Comment by OlivierSohn** [comment]
I also removed the use of LambdaCase to have it compile with older GHCs. 

CI is green now!

File: packunused.hs [hunk 1]
```diff
@@ -271,18 +287,21 @@ main = do
...
 
-    compBuildOrder = map fst . allComponentsInBuildOrder
-
 componentNameAndModules :: Bool -> Component -> (String, String, [ModuleName])
 componentNameAndModules addMainMod c  = (n, n2, m)
   where
     m = nub $ sort $ m0 ++ PD.otherModules (componentBuildInfo c)
 
     (n, n2, m0) = case c of
         CLib   ci -> ("library", "", PD.exposedModules ci)
```
Comment (hvr) [inline packunused.hs]: I'm missing something here that extracts the unqualComponentName of sub-libs; see http://hackage.haskell.org/package/Cabal-2.2.0.1/docs/Distribution-Types-Library.html#v:libName

Comment (OlivierSohn) [reply packunused.hs]: It's not missing, it's moved to line 154

Comment (hvr) [reply packunused.hs:296]: No :-)

**Pull request merged**